packages/flutter/test/widgets/scrollbar_test.dart

```diff
@@ -0,0 +1,435 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/src/physics/utils.dart' show nearEqual;
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+const Color _kScrollbarColor = Color(0xFF123456);
+const double _kThickness = 2.5;
+const double _kMinThumbExtent = 18.0;
+
+CustomPainter _buildPainter({
+  TextDirection textDirection = TextDirection.ltr,
+  EdgeInsets padding = EdgeInsets.zero,
+  Color color = _kScrollbarColor,
+  double thickness = _kThickness,
+  double mainAxisMargin = 0.0,
+  double crossAxisMargin = 0.0,
+  Radius radius,
+  double minLength = _kMinThumbExtent,
+  double minOverscrollLength = _kMinThumbExtent,
+  ScrollMetrics scrollMetrics,
+}) {
+  return ScrollbarPainter(
+    color: color,
+    textDirection: textDirection,
+    thickness: thickness,
+    padding: padding,
+    mainAxisMargin: mainAxisMargin,
+    crossAxisMargin: crossAxisMargin,
+    radius: radius,
+    minLength: minLength,
+    minOverscrollLength: minOverscrollLength,
+    fadeoutOpacityAnimation: kAlwaysCompleteAnimation,
+  )..update(scrollMetrics, scrollMetrics.axisDirection);
+}
+
+class _DrawRectOnceCanvas implements Canvas {
```

Comment:
For what it's worth, you ended up reimplementing mockito https://github.com/dart-lang/mockito :) which we use elsewhere in tests (if you search for the word mockito). 

You don't have to change since it's more or less the same length of code for your specific scenario but typically it's easier to use than to spin your own mocks.